rust/src/controllers/invitations.rs

```diff
@@ -85,20 +87,28 @@ pub fn notify_accepted(
             ON inv.id = $1 AND inv.recipient_username = recipient_user_details.username
          INNER JOIN user_details creator_user_details
             ON inv.id = $1 AND inv.creator_username = creator_user_details.username",
-        &[&data.uuid],
-    )
-    .and_then(|row| {
-        Ok(AcceptedNotificationData {
-            creator: row.get("creator"),
-            language: row.get("lang"),
-            recipient: row.get("first_name"),
+            &[&data.uuid],
+        )
+        .and_then(|row| {
+            Ok(AcceptedNotificationData {
+                creator: row.get("creator"),
+                language: row.get("lang"),
+                recipient: row.get("first_name"),
+            })
         })
-    })
-    .map_err(|w| APIInternalError {
-        msg: TranslationIds::BackendIssue,
-        engineering_error: Some(w.to_string()),
-    })
-    .and_then(|inv_data| push_accepted_notification(conn, channel, &inv_data))
+        .map_err(|w| APIInternalError {
+            msg: TranslationIds::BackendIssue,
+            engineering_error: Some(w.to_string()),
+        })
+        .and_then(|inv_data| {
+            let channel = RabbitConnection::insecure_open(&get_rabbitmq_uri())
+                .and_then(|mut connection| connection.open_channel(None))
+                .unwrap();
+
+            let result = push_accepted_notification(conn, &channel, &inv_data);
+            let _channel_close = channel.close();
+            return result;
+        });
```

Comment:
```suggestion
        .and_then(|inv_data| {
            RabbitConnection::insecure_open(&get_rabbitmq_uri())
                .and_then(|mut connection| connection.open_channel(None)).
                map_or_else(|_| Ok(()), |channel| {
                       let result = push_accepted_notification(conn, &channel, &inv_data);
                       let _ = channel.close();
                       result
                })            
        });
```

Reply:
I guess that we should avoid unwrapping when opening the connection so if it fails just do nothing (return Ok(()))

About the map_or_else function here: https://doc.rust-lang.org/std/result/enum.Result.html#method.map_or_else

Reply:
Yes, this request was not ready for review :)

Reply:
Hey @JasterV we would like to catch the errors as they occur so that we can diagnose rabbitmq